File: src/main/java/no/nav/familie/ks/sak/app/behandling/domene/grunnlag/søknad/SøknadGrunnlagRepository.java
package no.nav.familie.ks.sak.app.behandling.domene.grunnlag.søknad;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.Optional;

public interface SøknadGrunnlagRepository extends JpaRepository<SøknadGrunnlag, Long> {

    @Query(value="SELECT s FROM SøknadGrunnlag s WHERE s.behandlingId = :behandlingsId and s.aktiv = true")
    Optional<SøknadGrunnlag> finnGrunnlag(Long behandlingsId);
}

File: src/main/kotlin/no/nav/familie/ks/sak/kjerne/søknad/domene/SøknadGrunnlagRepository.kt
package no.nav.familie.ks.sak.kjerne.søknad.domene

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface SøknadGrunnlagRepository : JpaRepository<SøknadGrunnlag, Long> {

    @Query("SELECT gr FROM SøknadGrunnlag gr WHERE gr.behandlingId = :behandlingId AND gr.aktiv = true")
    fun hentAktiv(behandlingId: Long): SøknadGrunnlag?

    @Query("SELECT gr FROM SøknadGrunnlag gr WHERE gr.behandlingId = :behandlingId")
    fun hentAlle(behandlingId: Long): List<SøknadGrunnlag>
}
